pkgs/development/libraries/ffmpeg/6.nix

```diff
@@ -0,0 +1,13 @@
+import ./generic.nix rec {
+  version = "6.0";
+  sha256 = "sha256-RVbgsafIbeUUNXmUbDQ03ZN42oaUo0njqROo7KOQgv0=";
+
+  extraPatches = [
+    # this can be removed post 6.1
+    {
+      name = "fix_aacps_tablegen";
+      url = "https://git.ffmpeg.org/gitweb/ffmpeg.git/patch/814178f92647be2411516bbb82f48532373d2554";
+      hash = "sha256-FQV9/PiarPXCm45ldtCsxGHjlrriL8DKpn1LaKJ8owI=";
+    }
+  ];
```

Comment:
This is fine and I built the infra to have this attr but perhaps we might want to put version-specific patches into generic.nix, guarded behind an `optionals` obviously.

Reply:
I have pulled the patch into generic.nix, which looks fine for now, but could become hard to read if there were longer-lived patches applying to only specific major versions (this one is short-lived).

Reply:
What I don't particularly like about this, is that if you now do `ffmpeg_6.override { nv-codec-headers = nv-codec-headers-12; }`, it looks like it should do something, but since that argument is ignored in favor of nv-codec-headers-11... it won't do anything. :/

Reply:
Okay, if you think it's fine to assume that people will read the expression before overriding. Adding a comment isn't a clear improvement because it can become out-of-sync with the version selection logic below, and if you're already looking at the file, searching for the next occurrence of nv-codec isn't much of a jump.